Update blink for new service calls (#30267)
home-assistant/home-assistant.github.io: source/_integrations/blink.markdown

## Services

> Any sequential calls to {% term services %}  relating to blink should have a minimum of a 5 second delay in between them to prevent the calls from being throttled and ignored.  Services have targets.

Reviewer: ```suggestion
Any sequential calls to {% term services %}  relating to blink should have a minimum of a 5 second delay in between them to prevent the calls from being throttled and ignored. The services that act on a camera needs a target parameter.
```

> Any sequential calls to {% term services %}  relating to blink should have a minimum of a 5 second delay in between them to prevent the calls from being throttled and ignored. The services that act on a camera needs a target parameter.

Reviewer: ```suggestion
Any sequential calls to {% term services %} relating to blink should have a minimum of a 5 second delay in between them to prevent the calls from being throttled and ignored. The services that act on a camera needs a target parameter.
```

## Services

> Any sequential calls to {% term services %}  relating to blink should have a minimum of a 5 second delay in between them to prevent the calls from being throttled and ignored.  The services that act on a camera needs a target parameter.

Reviewer: ```suggestion
Any sequential calls to {% term services %}  relating to blink should have a minimum of a 5 second delay in between them to prevent the calls from being throttled and ignored. The services that act on a camera needs a target parameter.
```

### `blink.blink_update`

Force a refresh of the Blink system.
This service is depreciated, use `Home Assistant Core Integration: Update entity`
| Service Data Attribute | Optional | Description                            |
| ---------------------- | -------- | -------------------------------------- |

> | `config_entry_id`      | no       | Blink config to update.                |

Reviewer 2: We should remove deprecated services from our documentation.

### `blink.trigger_camera`

Trigger a camera to take a new still image.

| Service Data Attribute | Optional | Description                            |
| ---------------------- | -------- | -------------------------------------- |

> | `entity_id`            | no.      | Camera entity to take picture with.    |

Reviewer 2: This change is unrelated, and also incorrect. This service seem to be an entity service, meaning it can be targeted without an entity ID: thus optional.

### `blink.save_video`

Save the last recorded video of a camera to a local file. Note that in most cases, Home Assistant will need to know that the directory is writable via the `allowlist_external_dirs` in your `configuration.yaml` file (see example below).

| Service Data Attribute | Optional | Description                              |
| ---------------------- | -------- | ---------------------------------------- |

> | `entity_id`            | no       | Name of camera containing video to save. |

Reviewer 2: same

Author: Thanks - I missed these after the recommended changes to the PR.

| Service Data Attribute | Optional | Description                              |
| ---------------------- | -------- | ---------------------------------------- |
| `file_path`            | no       | Location of save files.                  |

### `blink.send_pin`

Send a new pin to blink.  Since Blink's 2FA implementation is new and changing, this is to allow the integration to continue to work with user intervention.  The intent is to handle all of this behind the scenes, but until the login implementation is settled this was added.  To use it, you simply call the service with the pin you receive from Blink as the payload (for a simple "Allow this Device" email, you may keep the `pin` value empty).

| Service Data Attribute | Optional | Description                  |
| ---------------------- | -------- | ---------------------------- |
| `config_entry_id`      | no       | Blink config to send pin to. |
| `pin`                  | no       | 2FA Pin received from blink. |

### Other services

> In addition to the services mentioned above, there are generic `camera`, `alarm_control_panel`, and `homeassistant` services are available. The `camera.enable_motion_detection` and `camera.disable_motion_detection` services allow for individual cameras to be enabled and disabled, respectively, within the Blink system. The `alarm_control_panel.alarm_arm_away` and `alarm_control_panel.alarm_disarm` services allow for the whole system to be armed and disarmed, respectively. The `homeassistant.update_entity` service will force an update of the blink system.  Blink Mini cameras linked to an existing sync module cannot be armed/disarmed individually via Home Assistant.

Reviewer: ```suggestion
In addition to the services mentioned above, there are generic `camera`, `alarm_control_panel`, and `homeassistant` services available. The `camera.enable_motion_detection` and `camera.disable_motion_detection` services allow for individual cameras to be enabled and disabled, respectively, within the Blink system. The `alarm_control_panel.alarm_arm_away` and `alarm_control_panel.alarm_disarm` services allow for the whole system to be armed and disarmed, respectively. The `homeassistant.update_entity` service will force an update of the blink system. Blink Mini cameras linked to an existing sync module cannot be armed/disarmed individually via Home Assistant.
```

> In addition to the services mentioned above, there are generic `camera`, `alarm_control_panel`, and `homeassistant` services are available. The `camera.enable_motion_detection` and `camera.disable_motion_detection` services allow for individual cameras to be enabled and disabled, respectively, within the Blink system. The `alarm_control_panel.alarm_arm_away` and `alarm_control_panel.alarm_disarm` services allow for the whole system to be armed and disarmed, respectively.  The `homeassistant.update_entity` service will force an update of the blink system.  Blink Mini cameras linked to an existing sync module cannot be armed/disarmed individually via Home Assistant.

Reviewer: ```suggestion
In addition to the services mentioned above, there are generic `camera`, `alarm_control_panel`, and `homeassistant` services are available. The `camera.enable_motion_detection` and `camera.disable_motion_detection` services allow for individual cameras to be enabled and disabled, respectively, within the Blink system. The `alarm_control_panel.alarm_arm_away` and `alarm_control_panel.alarm_disarm` services allow for the whole system to be armed and disarmed, respectively. The `homeassistant.update_entity` service will force an update of the blink system.  Blink Mini cameras linked to an existing sync module cannot be armed/disarmed individually via Home Assistant.
```

| Service Data Attribute | Optional | Description                  |
| ---------------------- | -------- | ---------------------------- |

> | `config_entry_id`      | no       | Blink config to send pin to. |

Reviewer 2: same

Author: This one currently is not a target, we are only pointing at the config_entry via a selector.  Is there a different way to point at a config entry?

Reviewer 2: No, this was just my review mistake, sorry about that 👍

Author: I made plenty of those myself.  Learning new stuff everyday!

  action:
    service: blink.save_video
    data:

>       entity_id: "My Camera"

Reviewer 2: This seems wrong? Entity ID should not be part of the data but part of the target?

  action:
    - service: blink.save_recent_clips
      data:

>         entity_id: My Camera

Reviewer 2: same?

Reviewer: "My Camera" isn't an entity_id.
